src/directive/permissions.directive.ts

```diff
@@ -52,70 +56,77 @@ export class NgxPermissionsDirective implements OnInit, OnDestroy {
-                    this.validateOnlyPermissions();
-                } else {
-                    this.handleAuthorisedPermission(this.ngxPermissionsExceptThen || this.ngxPermissionsThen || this.templateRef )
-                }
+        Promise.all([ this.permissionsService.hasPermission(this.ngxPermissionsExcept), this.rolesService.hasOnlyRoles(this.ngxPermissionsExcept) ])
+               .then(([ hasPermission, hasRole ]) => {
+                   if (hasPermission || hasRole) {
+                       this.handleUnauthorisedPermission(this.ngxPermissionsExceptElse || this.ngxPermissionsElse);
+                   } else {
+                       if (!!this.ngxPermissionsOnly) {
+                           throw false;
+                       } else {
+                           this.handleAuthorisedPermission(this.ngxPermissionsExceptThen || this.ngxPermissionsThen || this.templateRef);
+                       }
+                   }
+               }).catch(() => {
+            if (!!this.ngxPermissionsOnly) {
+                this.validateOnlyPermissions();
+            } else {
+                this.handleAuthorisedPermission(this.ngxPermissionsExceptThen || this.ngxPermissionsThen || this.templateRef);
+            }
         });
     }
 
     private validateOnlyPermissions(): void {
-         Promise.all([this.permissionsService.hasPermission(this.ngxPermissionsOnly), this.rolesService.hasOnlyRoles(this.ngxPermissionsOnly)])
-            .then(([permissionPr,  roles]) => {
-                if (permissionPr || roles) {
-                   this.handleAuthorisedPermission(this.ngxPermissionsOnlyThen || this.ngxPermissionsThen || this.templateRef)
-                } else {
-                    this.handleUnauthorisedPermission(this.ngxPermissionsOnlyElse || this.ngxPermissionsElse);
-                }
-            }).catch(() => {
-                this.handleUnauthorisedPermission(this.ngxPermissionsOnlyElse || this.ngxPermissionsElse);
-        })
+        Promise.all([ this.permissionsService.hasPermission(this.ngxPermissionsOnly), this.rolesService.hasOnlyRoles(this.ngxPermissionsOnly) ])
+               .then(([ permissionPr, roles ]) => {
+                   if (permissionPr || roles) {
+                       this.handleAuthorisedPermission(this.ngxPermissionsOnlyThen || this.ngxPermissionsThen || this.templateRef);
+                   } else {
+                       this.handleUnauthorisedPermission(this.ngxPermissionsOnlyElse || this.ngxPermissionsElse);
+                   }
+               }).catch(() => {
+            this.handleUnauthorisedPermission(this.ngxPermissionsOnlyElse || this.ngxPermissionsElse);
+        });
     }
 
-
     private handleUnauthorisedPermission(template: TemplateRef<any>): void {
-        this.permissionsUnauthorized.emit();
-        this.showTemplateBlockInView(template);
+        if (!isBoolean(this.currentAuthorizedState) || this.currentAuthorizedState) {
```

Comment:
And here probably remove isBoolean property.

Reply:
You must have a flag indicating the first rendering. Otherwise, the initialization will not work correctly with unauthorised permission.

Reply:
Yes bless tests :-)